### ISSUE TITLE
More cycles on guest OS = capped speed?

### ISSUE THREAD

**Issue by Xbox-Series-X-User** [schellingb/dosbox-pure#484]
I'm running Windows 98 SE  (issue is still there when not running anything just the core by itself) and it seems anything passed 300mhz gets limited.( 1.2 ghz athelon goes ~7-15fps, Pentium III 600mhz goes ~30) I can use the fast forward function on retroarch to maintain 60+ fps (200+ unlimited fast forward speed) but if it can reach a stable 60 with fast forward on, why isn't it by default? Also I have been unable to figure out how to get networking to work on the latest nightly build of retroarch (Xbox only issue?)  but the performance is what I'm curious about. Thanks for the awesome user friendly core. I hope you keep up the amazing work and have a great day.

**Comment by Xbox-Series-X-User** [comment]
![17105752408506445437488547688045](https://github.com/schellingb/dosbox-pure/assets/86686097/014fad3a-e44a-49c6-9891-5acc561017d2)
Sorry if the picture is a bit blurry taking screenshots in retroarch doesn't show the performance information

**Comment by Xbox-Series-X-User** [comment]
![17105754895417373778368445411689](https://github.com/schellingb/dosbox-pure/assets/86686097/d645cfdd-7cf0-47cc-a512-85b9a134f06e)

**Comment by schellingb** [comment]
Once you ask DOSBox to emulate more cycles than your host system can manage to calculate in real-time, things will get very bad. So if you see "Speed: 10%" this basically means you are trying to emulate 10 times more the CPU cycles than the host system can handle.

During fast forwarding, the emulated cycles are set to a fixed rate. 10000 while in real mode and 30000 while in protected mode. Protected mode is a different operational mode supported fully by Intel 386 and newer CPUs. Most DOS games made after 1993 or so use this, and so do all Windows since 3.1. While fast forwarding, I chose those fixed cycle counts because it gave the best results with a few tests I did. If one fast forwards with too many emulated cycles, an older game might take a long time to actually draw a frame because so much time is spent emulating the CPU essentially doing nothing. The goal of fast forwarding certainly isn't to make the cycles number go high, it is to make an emulated game run as fast as possible. For most DOS games this means keeping the count low but ignore the frequency of the emulated monitor and just let it run and draw frames as fast as possible.
